### DISCUSSION TITLE
Adding Zoho as Authentication provider

### DISCUSSION THREAD

**Discussion by siddharthakhuntia-lohum** [outline/outline#6517] [Ideas]
I am currently using your repository for my project and it's been a great help. However, my company uses Zoho for workspace management, which provides authentication through OIDC. I noticed that your repository supports several major authentication providers, but Zoho isn't one of them.

In the current codebase, I need to change the code from profile.email to profile.Email  in `/plugins/oidc/server/auth/oidc.ts` to make it compatible with Zoho's authentication. Because of this, I can't use your provided image and have to create my own on top of your repository.

I believe adding Zoho as an authentication provider would be beneficial not only for my company but also for other organizations that use Zoho. It would save us from having to modify the codebase and create our own images, and it would make your repository even more versatile.

I hope you'll consider this request. If you need more information about Zoho's OIDC authentication, I'd be happy to provide it.

Thank you for your time and consideration.

**Comment by tommoor** [comment]
The point of OIDC is that it is standardized and no more individual auth providers need to be added. Looking at Zoho's own documentation that parameter should be lowercased (as the standard):

https://www.zoho.com/accounts/protocol/oauth/sign-in-using-zoho.html

<img width="408" alt="image" src="https://github.com/outline/outline/assets/380914/e1398367-5705-43d0-896d-a28522b59fb1">

Definitely weird that you would need to patch in this way. Perhaps a question for Zoho?

**Comment by siddharthakhuntia-lohum** [comment]
Thanks for taking the time to reply. Upon taking a closer look at Zoho services, I found that the issue lies at Zoho's end.
